lib/checks/checks.go

```diff
@@ -756,28 +756,56 @@ func checkRAM(info ServerInfo, ram schema.RAM) error {
 	return nil
 }
 
-// checkSameOS makes sure all servers have the same OS/version
+// checkSameOS verifies the OS distribution requirement for the specified set of servers.
+// The check will pass if all nodes in the cluster are based on the same OS distribution and major version.
+// Variance in minor/patch versions is acceptable.
 func checkSameOS(servers []Server) error {
-	osToNodes := make(map[string][]string)
+	// distros maps distribnution name to list of versions
+	distros := make(map[string][]string)
 	for _, server := range servers {
-		os := systeminfo.OS(server.GetOS()).Name()
-		osToNodes[os] = append(osToNodes[os], fmt.Sprintf("%v (%v)",
-			server.ServerInfo.GetHostname(), server.AdvertiseAddr))
+		info := server.GetOS()
+		distros[info.ID] = append(distros[info.ID], info.Version)
 	}
+	if len(distros) != 1 {
+		return trace.BadParameter("servers have different OS distributions: %v", formatKeysAsList(distros))
+	}
+	// Version verification is purposedly simply and will compare the prefixes
+	// upto to either the first '.' or eol
+	for _, versions := range distros {
+		if !verifyCommonVersionPrefix(versions...) {
+			return trace.BadParameter("servers have different OS versions: %v", formatAsList(distros))
+		}
+	}
+	log.Info("Servers passed check for the same OS.")
+	return nil
+}
 
-	if len(osToNodes) > 1 {
-		var formatted []string
-		for os, nodes := range osToNodes {
-			formatted = append(formatted, fmt.Sprintf(
-				"%v: %v", os, strings.Join(nodes, ", ")))
+func verifyCommonVersionPrefix(versions ...string) bool {
+	prefixes := make(map[string]struct{})
+	for _, v := range versions {
+		index := strings.Index(v, ".")
```

Comment:
It might be just a question of preference of course, but why mess with indices/lengths/maps/etc. if all we need to do is compare first elements of strings split by a dot?

```go
func verifyCommonVersionPrefix(versions ...string) bool {
  if len(versions) <= 1 {
    return true
  }
  for i := 0; i < len(versions)-1; i += 1 {
    if strings.Split(versions[i], ".") != strings.Split(versions[i+1], ".") {
      return false
    }
  }
  return true
}
```